tests/global-setup.ts

```diff
@@ -0,0 +1,23 @@
+// global-setup.ts
+import { chromium, FullConfig } from '@playwright/test';
+import { LoginPage } from './page-object-model/loginpage';
+
+async function globalSetup(config: FullConfig) {
+    const browser = await chromium.launch();
+    //######### TODO: Replace this with config ###############
+    const page = await browser.newPage({ baseURL: 'http://localhost:4200' });
+
+    // This saves a clean slate which we can use in tests that require us not to be logged in
+    await page.context().storageState({ path: 'cleanStorageState.json' });
+
+    let loginPage = new LoginPage(page);
+    await loginPage.navigateTo();
+    await loginPage.fillFormWithValidDetails();
+    await loginPage.submitForm();
+
```

Comment:
Suggestion: In our use case we have a separate dashboard api for handling auth (among other things) If we were to use this approach it would mean that we would have to be integrated with a dashboard api in order to test the UI (i.e we wouldn't be testing our UI in isolation). 

Might be another PR after this one but could be worth including an example where we just set the local storage. Couple of ways of doing this ;
Using the a setlocal storage package;
```
public async SetStorageToken(storageToken: string, userName: string){
     await this.page.evaluate(()=> {
          localStorage.setItem("authenticated", "true");
        
      }
    );
  }
```
or using page route;
```
public setTokenRoute(page, expectedStatus) { 
    page.route('**/token', route => {
      route.fulfill({
        status: expectedStatus,
        body: JSON.stringify({ "authenticated": "true"})
      });
    });
  }
```

Reply:
Worth noting as well that I think storing auth in local storage is not best pratise (briefly remember reading it somewhere) but don't know enough about UI/SPA's to know what the correct approach is :S

Reply:
What's the saying a little bit of knowledge is dangerous :D and I can't find the thing I read like a few months ago so I agree think we should leave the local storage bit!